src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figuresList = new Figure[(int) (Math.random() * 23)];
+        for (int i = 0; i < figuresList.length; i++) {
+            figuresList[i] = new FigureSupplier().randomFigure();
```

Comment:
new FigureSupplier is created on each loop iteration

Reply:
done